### PR TITLE
feat: Add the disconnect and change world buttons to the character selection screen

### PR DESCRIPTION
**Pull request by sofiedotcafe**
Add the disconnect and change world buttons to the character selector screen 
Screenshot_20230630_112727

### CODE REVIEW

File: common/src/main/java/com/wynntils/screens/characterselector/widgets/ChangeWorldButton.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import com.wynntils.utils.render.RenderUtils;
+import com.wynntils.utils.render.Texture;
+import java.util.List;
+import net.minecraft.ChatFormatting;
+import net.minecraft.client.gui.screens.inventory.tooltip.ClientTooltipComponent;
+import net.minecraft.network.chat.Component;
+
+public class ChangeWorldButton extends WynntilsButton {
+
+    private static final String LOBBY_SERVER = "lobby.wynncraft.com";
```
Comment (ryanzhoudev) [inline common/src/main/java/com/wynntils/screens/characterselector/widgets/ChangeWorldButton.java]: What is this for?

File: common/src/main/java/com/wynntils/screens/characterselector/widgets/ChangeWorldButton.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                this.width,
+                this.height,
+                0,
+                0,
+                Texture.CHANGE_WORLD_BUTTON.width(),
+                Texture.CHANGE_WORLD_BUTTON.height() / 2,
+                Texture.CHANGE_WORLD_BUTTON.width(),
+                Texture.CHANGE_WORLD_BUTTON.height());
+
+        if (isHovered) {
```
Comment (ryanzhoudev) [inline common/src/main/java/com/wynntils/screens/characterselector/widgets/ChangeWorldButton.java:55]: Can we pull this if clause out to some method? It's also used in DisconnectButton and PlayButton

File: common/src/main/java/com/wynntils/screens/characterselector/widgets/ChangeWorldButton.java (hunk 1 above)
Comment (sofiedotcafe) [reply common/src/main/java/com/wynntils/screens/characterselector/widgets/ChangeWorldButton.java]: Oh the LOBBY_SERVER?

I can remove it

File: common/src/main/java/com/wynntils/screens/characterselector/widgets/ChangeWorldButton.java (hunk 2 above)
Comment (sofiedotcafe) [reply common/src/main/java/com/wynntils/screens/characterselector/widgets/ChangeWorldButton.java:55]: Yeah, maybe we could

Comment (sofiedotcafe) [reply common/src/main/java/com/wynntils/screens/characterselector/widgets/ChangeWorldButton.java:55]: Maybe like:
```java
private void renderTooltip(PoseStack poseStack, int mouseX, int mouseY, List<Component> tooltip) {
    poseStack.pushPose();
    List<ClientTooltipComponent> clientTooltipComponents =
            TooltipUtils.componentToClientTooltipComponent(tooltip);
    poseStack.translate(
            mouseX
                    - TooltipUtils.getToolTipWidth(
                            clientTooltipComponents, FontRenderer.getInstance().getFont()),
            mouseY - TooltipUtils.getToolTipHeight(clientTooltipComponents),
            100);
    RenderUtils.drawTooltip(poseStack, tooltip, FontRenderer.getInstance().getFont(), true);
    poseStack.popPose();
}
```

File: common/src/main/java/com/wynntils/screens/characterselector/CharacterSelectorScreen.java [hunk 3]
```diff
@@ -112,6 +114,24 @@ protected void doInit() {
                 (int) addButtonHeight,
                 this));
 
+        float disconnectButtonWidth = Texture.DISCONNECT_BUTTON.width() * currentTextureScale;
+        float disconnectButtonHeight = Texture.DISCONNECT_BUTTON.height() / 2f * currentTextureScale;
+        this.addRenderableWidget(new DisconnectButton(
+                (int) (this.width - disconnectButtonWidth - 10f),
+                (int) (10f),
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/screens/characterselector/CharacterSelectorScreen.java]: Notice how every other Y coordinate is based on height while this is hard coded, meaning it won't work on all resolutions. Please fix this

File: common/src/main/java/com/wynntils/screens/characterselector/widgets/ChangeWorldButton.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+import com.wynntils.utils.render.FontRenderer;
+import com.wynntils.utils.render.RenderUtils;
+import com.wynntils.utils.render.Texture;
+import java.util.List;
+import net.minecraft.ChatFormatting;
+import net.minecraft.client.gui.screens.inventory.tooltip.ClientTooltipComponent;
+import net.minecraft.network.chat.Component;
+
+public class ChangeWorldButton extends WynntilsButton {
+
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/screens/characterselector/widgets/ChangeWorldButton.java]:
```suggestion
```

File: common/src/main/java/com/wynntils/screens/characterselector/widgets/ChangeWorldButton.java (hunk 2 above)
Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/screens/characterselector/widgets/ChangeWorldButton.java:55]: I know the concern is code duplication but sharing this code would mean adding a superclass which might not end up cleaner than copying this once.

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/screens/characterselector/widgets/ChangeWorldButton.java:55]: But I can also see that basically the whole class is duplicated... Up to @DonkeyBlaster if they want it refactored or if it is fine this way.

File: common/src/main/java/com/wynntils/screens/characterselector/CharacterSelectorScreen.java (hunk 3 above)
Comment (sofiedotcafe) [reply common/src/main/java/com/wynntils/screens/characterselector/CharacterSelectorScreen.java]: Ok, i will, but isn't the PlayButton using basically the same thing? @kristofbolyai

Comment (sofiedotcafe) [reply common/src/main/java/com/wynntils/screens/characterselector/CharacterSelectorScreen.java]: Also isn't the X also affected by it?

Comment (sofiedotcafe) [reply common/src/main/java/com/wynntils/screens/characterselector/CharacterSelectorScreen.java]: Fixed with the commit [781dde9](https://github.com/Wynntils/Artemis/pull/1650/commits/781dde999d7fd791393e8b0548cab6e3bf7990c7)

**Pull request closed without merging**